pkg/authorization/util/util.go

```diff
@@ -46,3 +65,65 @@ func Authorize(sarClient internalversion.SubjectAccessReviewInterface, user user
 	}
 	return kerrors.NewForbidden(schema.GroupResource{Group: resourceAttributes.Group, Resource: resourceAttributes.Resource}, resourceAttributes.Name, err)
 }
+
+func NewReadOnlyRuleResolver(policyRegistry policyregistry.Registry, policyBindingRegistry policybindingregistry.Registry, clusterPolicyRegistry clusterpolicyregistry.Registry, clusterBindingRegistry clusterpolicybindingregistry.Registry) rulevalidation.AuthorizationRuleResolver {
+	return rulevalidation.NewDefaultRuleResolver(
+		&policyregistry.ReadOnlyPolicyListerNamespacer{
+			Registry: policyRegistry,
+		},
+		&policybindingregistry.ReadOnlyPolicyBindingListerNamespacer{
+			Registry: policyBindingRegistry,
+		},
+		&clusterpolicyregistry.ReadOnlyClusterPolicyClientShim{
+			ReadOnlyClusterPolicy: clusterpolicyregistry.ReadOnlyClusterPolicy{Registry: clusterPolicyRegistry},
+		},
+		&clusterpolicybindingregistry.ReadOnlyClusterPolicyBindingClientShim{
+			ReadOnlyClusterPolicyBinding: clusterpolicybindingregistry.ReadOnlyClusterPolicyBinding{Registry: clusterBindingRegistry},
+		},
+	)
+}
+
+func GetAuthorizationStorage(optsGetter restoptions.Getter, cachedRuleResolver rulevalidation.AuthorizationRuleResolver) (
+	policyregistry.Storage,
```

Comment:
> Can I buy a struct?

This is likely to live past the authorizer and storage because its used by client tools which need to be compatible with old clusters.  Since it lives beyond 3.6, I'd like something a little prettier.